src/project/network/Group.java

```diff
@@ -9,6 +9,9 @@
     private int consumption;
     private int originalconsumption; // a manipulé avec le facteur
     private SubStation station;
+    private static ConsumptionMacro consumpMacro;
```

Comment:
Pourquoi statique ici si tu en fais un membre non-statique de Network  (ou pourquoi en faire un membre de Network s'il est accessible en membre statique de Group) ?

Reply:
Comme la localisation dans le code  du test, du lancement et du run reste un peu flou pour moi, je n'étais pas certain que Network contiendrait ConsumpMacro. Ca dépends de comment le reste est fait, en soi la seule utilité de le mettre dans Network c'est si la fonction run y a lieu, afin d'incrémenter le curseur.

Reply:
En fait le truc c'est que je l'ai mis dans Network pour pouvoir y accéder facilement en dehors d'un 
"for nodes n" où il aurait été plus compliqué de lancer l'avancement du curseur une seule fois.

Reply:
Sauf que si l'instance de ConsumptionMacro est un membre statique de Group tu peux y accéder à tout endroit en faisant `Group.consumpMacro` pour peu que le membre ne soit pas privé (ou utiliser une méthode si c'est important de garder consumpMacro privé). Après oui, pourquoi pas, mais comme j'ai dit plus haut si consumpMacro est statique tu ne dois pas y accéder en faisant `this.consumpMacro`.